packages/widget-v2/src/components/RenderWalletList.tsx

```diff
@@ -54,71 +53,130 @@ export const RenderWalletListHeader = ({
+      rightContent,
+    } = wallet;
+    return (
+      <ModalRowItem
+        key={walletName}
+        onClick={() => {
+          connectMutation.mutate(wallet);
+        }}
+        style={{ marginTop: ITEM_GAP }}
+        leftContent={
+          <Row align="center" gap={10}>
+            {imageUrl && (
+              <img
+                height={35}
+                width={35}
+                style={{ objectFit: "cover" }}
+                src={imageUrl}
+                alt={`${walletPrettyName} logo`}
+              />
+            )}
+
+            <Text>{walletPrettyName}</Text>
+          </Row>
+        }
+        rightContent={rightContent?.()}
+      />
+    );
+  };
 
   const height = useMemo(() => {
     return Math.min(530, walletList.length * (ITEM_HEIGHT + ITEM_GAP));
   }, [walletList]);
 
+  const container = useMemo(() => {
+    switch (true) {
+      case !!connectMutation.isError:
+        return (
+          <StyledInnerContainer height={height}>
+            <StyledLoadingContainer>
+              <StyledAnimatedBorder
+                width={80}
+                height={80}
+                backgroundColor={theme.primary.text.normal}
+                txState="failed"
+                borderSize={8}
+              >
+                <img
+                  style={{ objectFit: "cover" }}
+                  src={connectMutation.variables?.walletInfo.logo}
+                  alt={`${connectMutation.variables?.walletPrettyName} logo`} />
+              </StyledAnimatedBorder>
+              <Text color={theme.primary.text.lowContrast}>Failed to connect to {connectMutation.variables?.walletPrettyName}</Text>
+            </StyledLoadingContainer>
+          </StyledInnerContainer>
+        );
+      case !!connectMutation.isPending:
+        return (
+          <StyledInnerContainer height={height}>
+            <StyledLoadingContainer>
+              <StyledAnimatedBorder
+                width={80}
+                height={80}
+                backgroundColor={theme.primary.text.normal}
+                txState="broadcasted"
+                borderSize={8}
+              >
+                <img
+                  style={{ objectFit: "cover" }}
+                  src={connectMutation.variables?.walletInfo.logo}
+                  alt={`${connectMutation.variables?.walletPrettyName} logo`} />
+              </StyledAnimatedBorder>
+              <Text color={theme.primary.text.lowContrast}>Connecting to {connectMutation.variables?.walletPrettyName}</Text>
+            </StyledLoadingContainer>
+          </StyledInnerContainer>
+        );
+      default:
+        return <VirtualList
+          listItems={walletList}
+          height={height}
+          itemHeight={ITEM_HEIGHT + ITEM_GAP}
+          renderItem={renderItem}
+          itemKey={(item) => item.walletName}
+        />;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [connectMutation, height, walletList]);
+
```

Comment:
i think in this case we just use if statements instead of switch

```ts
if (connectMutation.isError) {
  return ...
} 
if (connectMutation.isPending) {
  return ...
}
return ()
```